sorting/random_pivot_quick_sort.cpp

```diff
@@ -0,0 +1,306 @@
+/**
+ * @file
+ * @brief A cpp program for random pivot quick sort algorithm.
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation.